sandbox/grommet-app/src/pages/layouts/kinds/OpsRampDetail/OpsRampDetailTable.tsx

```diff
@@ -0,0 +1,207 @@
+    ).length;
+
+    setUp(upNodes);
+    setDown(downNodes);
+    setUndefinedState(undefinedNodes);
+    setUnknown(unknownNodes);
+    setTotal(nodes.length);
+    setResult(nodes);
+  }, []);
+
+  useEffect(() => {
+    if (!view) return;
+
+    let filteredData = opsRamp.nodes;
+    if (!('properties' in view)) {
+      setQuickFilter('');
+    }
+
+    if (quickFilter) {
+      filteredData = filteredData.filter(nodes => nodes.state === quickFilter);
+    }
+    if (view && view.properties && typeof view.properties === 'object') {
+      Object.keys(view.properties).forEach(property => {
+        if (view.properties && Array.isArray(view.properties[property])) {
+          filteredData = filteredData.filter(
+            nodes =>
+              view.properties &&
+              view.properties[property].includes(nodes[property]),
+          );
+        }
+      });
+    }
+
+    setResult(filteredData);
+    setTotal(filteredData.length);
+  }, [view, quickFilter]);
+
+  const columns: ColumnConfig<Node>[] = [
+    {
+      property: 'name',
+      pin: true,
+      primary: true,
+      header: 'Name',
+      render: datum => (
+        <Box align="center" direction="row" gap="xsmall">
+          {datum.state === 'up' ? (
+            <StatusGoodSmall color="status-ok" />
+          ) : datum.state === 'down' ? (
+            <StatusCriticalSmall color="status-critical" />
+          ) : datum.state === 'unknown' ? (
+            <StatusUnknownSmall color="status-unknown" />
+          ) : datum.state === 'undefined' ? (
+            <StatusPlaceholderSmall />
+          ) : null}
+          <Anchor
+            onClick={() => {
+              setShowResultDetails(true);
+            }}
+          >
+            {datum.name}
+          </Anchor>
+        </Box>
+      ),
+    },
+    {
+      property: 'ip address',
+      header: 'IP Address',
+      sortable: false,
+    },
+    {
+      property: 'make',
+      header: 'Make',
+      sortable: false,
+    },
+    {
+      property: 'model',
+      header: 'Model',
+      sortable: false,
+    },
+  ];
+
+  return (
+    <Data
+      data={result}
+      total={total}
+      filteredTotal={result.length}
+      defaultView={defaultView}
+      gap="medium"
+      properties={{
+        model: {
+          label: 'Model',
+          options: ['VirtualMachine', 'VMware Virtual Platform'],
+        },
+        state: {
+          label: 'State',
+          options: ['up', 'down', 'unknown', 'undefined'],
+        },
+      }}
+      view={view}
+      onView={setView}
+    >
+      <QuickFilters
+        value={quickFilter}
+        setValue={setQuickFilter}
+        counts={{ up, down, unknown, undefinedState }}
+      />
+      <Box>
+        <Toolbar>
+          <DataSearch />
+          <DataFilters layer />
+          <DataSort drop />
+          <Button kind="toolbar" icon={<Share />} />
+          <Button kind="toolbar" icon={<SettingsOption />} />
+        </Toolbar>
```

Comment:
Based on the design which has selectable rows, I don't see any way to apply any actions to the multiple selected rows, so I'm interpreting the "gear" icon to represent "actions."

Actions (of which "share" is also a type of action) should be placed to the right of the toolbar.


```suggestion
<Toolbar justify='between'>
          <Toolbar>
            <DataSearch />
            <DataFilters layer />
            <DataSort drop />
          </Toolbar>
          <Toolbar gap="xsmall">
            <Button kind="toolbar" icon={<Share />} tip="Share Kubernetes nodes" />
            <Button kind="toolbar" icon={<Actions />} tip="Actions" />
          </Toolbar>
        </Toolbar>
```

Reply:
I feel like thats a fair assessment